core/core.php

```diff
@@ -181,4 +185,104 @@ public function validate_badge_order($user_id)
 
 		return true;
 	}
+
+	/**
+	 * Gets the url to the user avatar.
+	 * Returns false on failure or if there is no avatar.
+	 *
+	 * @param array $user_data Avatar data from the user table
+	 * @return array|boolean
+	 */
+	public function get_avatar_data($user_data)
+	{
+		if (!$this->config['allow_avatar'])
+		{
+			return false;
+		}
+
+		$row = [
+			'avatar' => $user_data['user_avatar'],
+			'avatar_width' => $user_data['user_avatar_width'],
+			'avatar_height' => $user_data['user_avatar_height'],
+		];
+
+		$driver = $this->avatar_manager->get_driver($user_data['user_avatar_type']);
+
+		if (!$driver)
+		{
+			return false;
+		}
+
+		$avatar_data = $driver->get_data($row);
+
+		if ($user_data['user_avatar_type'] === 'avatar.driver.gravatar')
+		{
+			$avatar_data['src'] = $this->get_gravatar_url($row);
+		}
+		else if ($user_data['user_avatar_type'] === 'avatar.driver.upload')
+		{
+			$avatar_data['src'] = $this->get_upload_avatar_url($user_data['user_avatar']);
+		}
+		else if ($user_data['user_avatar_type'] === 'avatar.driver.local')
+		{
+			$avatar_data['src'] = $this->get_local_avatar_url($user_data['user_avatar']);
```

Comment:
Might add a comment that it is important that the key is named 'src'. The array returned from `get_data($row)` already has a `src` key, containing relative urls or e-mail-adresses that need to be overridden when using this approach (`get_data()` does not give us any data we do not already have retrieved or hard-coded here).

Reply:
Yes, you're right - the avatar driver is not needed. I left it in there only to support other drivers :thinking: but if we decide that we only support gravatar, upload and local then this code can be simplified.

Reply:
I see. IMO and since @pchote basically already +1 this, I'd say go with the comment or leave it as it is so that it does not need another review.